mitodl/open-discussions
# Send emails to subscribers on new comment

#### What are the relevant tickets?
Fixes #467 
Fixes #470

#### What's this PR do?
Adds comment notification emails when a comment or post you are subscribed to is commented on.

#### How should this be manually tested?
- Turn on the feature flag `FEATURE_COMMENT_NOTIFICATIONS`
- For two users A and B, ensure both have their comment settings (at `/api/v0/notification_settings/comments/`) set to `"immediate"`
- As user A create a post (you get a subscription for this).
- Comment on that post  as user B and verify user A gets the appropriate email
- Comment on that comment as user A and verify user B gets the appropriate notification email

#### Any background context you want to provide?

There was necessary refactoring of reddit/betamax fixtures up to the top level of the app. These are now conditionally turned on via the `pytest.mark.betamax` marker to provide backwards compatibility with existing cassettes as well as _not_ cause other tests to generate cassettes.

#### Screenshots (if appropriate)
See mockup [here](https://github.com/mitodl/open-discussions/issues/470#issuecomment-379017588)
